sys/net/application_layer/gcoap/dtls_session_management.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2020 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_gcoap
+ * @{
+ *
+ * @file
+ * @brief       Implementation of DTLS Session Management for Gcoap
+ *
+ * @author      János Brodbeck <janos.brodbeck@ml-pa.com>
+ */
+
+#include "net/gcoap/dtls_session_management.h"
+#include "mutex.h"
+#include "net/sock/util.h"
+#include "xtimer.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+typedef struct {
+    sock_udp_ep_t ep;
+    session_state_t state;
+    uint64_t last_used;
+} dsm_session_t;
+
+static void _ep_to_session(const sock_udp_ep_t *ep, sock_dtls_session_t *session);
+static int _find_session(const sock_udp_ep_t *ep, dsm_session_t **session);
+
+static mutex_t _lock;
+static dsm_session_t _sessions[CONFIG_GCOAP_DTLS_SESSIONS_MAX];
```

Comment:
```suggestion
static dsm_session_t _sessions[DTLS_PEER_MAX];
```

Reply:
The idea behind introducing `CONFIG_GCOAP_DTLS_SESSIONS_MAX` was that `DTLS_PEER_MAX` is tied to tinyDTLS. So in the future there could be other DTLS pkgs (e.g. wolfSSL) which possibly have a different variable for that. So supporting more DTLS pkgs would go quicker in the future. 

But `CONFIG_GCOAP_DTLS_SESSIONS_MAX` needs to be exact the same as the DTLS implementation uses internally (because as far as I understand, you cannot specify how many sessions are allowed per socket).